### ISSUE TITLE
Make downloads of models independent of X-Repo-Commit http header to allow download via proxy repositories 

### ISSUE THREAD

**Issue by h-filzer** [huggingface/huggingface_hub#1796]
**Situation**
When I try to download a model using the "snapshot_download" function, the download fails because my JFrog Artifactory doesn't send an X-Repo-Commit HTTP header which is madatory at he moment. Even when I specify the exact version of the model (commit SHA) using the "revision" parameter, the download still fails.
 
**Describe the solution you'd like**
 I would like the model download to be possible without relying on the X-Repo-Commit HTTP header send by anyone, to allow downloads through proxy repositories in environments where direct internet access is not possible.
I am a python rookie but i think the point is that the `HfFileMetadata` is stuffed with the Header [(Line in file_download.py)](https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/file_download.py#L1611) which is then verified to be present in [Line 1293](https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/file_download.py#L1239) which gave me the error as it is not present. 
I would like to have the X-Repo-Commit optional. If the header is present use it. If not use the revision attribute from the function call. If both are missing throw an error.
What Do u think? I could make a PR for this if this is the way to go? Or do i miss something while i am trying to download a snapshot?

**Here is an example of the failing request**
`export HF_ENDPOINT=https://huggingface-generic-remote.mycompany-artifactory-proxy.com/artifactory/huggingface-generic-remote`

```python
from huggingface_hub import snapshot_download
snapshot_download(
    repo_id="openai/whisper-tiny", revision="be0ba7c2f24f0127b27863a23a08002af4c2c279"
)
```

**Comment by Wauplin** [comment]
Hey @h-filzer, thanks for opening the issue. At the moment there is no plan to support any other platform than the Hugging Face Hub for remote models and we are planning to deprecate the `HF_ENDPOINT` environment variable soon. If you need to run scripts in an environment where direct internet access is not possible, we advice to download the weights once on an internal shared drive and then set [`HUGGINGFACE_HUB_CACHE`](https://huggingface.co/docs/huggingface_hub/package_reference/environment_variables#huggingfacehubcache) and [`HF_HUB_OFFLINE`](https://huggingface.co/docs/huggingface_hub/package_reference/environment_variables#hfhuboffline) as environment variables. The first one should point to the shared cache while the second one explicitely prevent your scripts to access https://huggingface.co/.

**Comment by h-filzer** [comment]
Hey @Wauplin, thanks for the quick reply. Unfortunately that's bad news for everyone who's company does not allow direct internet access through their infrastructure. Downloading the model on someones computer and  then pushing it to the required environment by hand is very cumber stone as this does not scale or be part of an automated ci/cd process. Companies invested in Proxies or caches for maven artifacts or docker repositories to come across this problem. It would be great if downloading models through products like JFrogs Artifactory will still be possible in the future which also saves a lot of bandwidth and CO2. Do you have a timeline when the deprecation of `HF_ENDPOINT` will happen?

**Comment by Wauplin** [comment]
Hey @h-filzer sorry for late reply. We stopped promoting `HF_ENDPOINT` in our documentation but we will still support it at least until `huggingface_hub<1.0` (that is not planned at the moment). Understood for the requirements you have however it will most probably not be in our roadmap yet. I will still keep this issue open and invite anyone else landing on it to comment if there is more demand about it.

**Comment by Adamwgoh** [comment]
hi @Wauplin One of the main use case of `HF_ENDPOINT` has been for using it as a pointer for remote artifact repositories so HfApi and cache resolution would work with them. For instance, JFrog now offer Huggingface proxy repository which relies on setting this variable to point to their cache repository endpoint instead (ie, `my-jfrog-instance.jfrog.io/my-hub-repo` ). Since they only started announcing support for it back in Sep '23, I'm a little surprised to read that it will be retired soon. If there is intention of retiring this variable, may I ask what would be a good alternative for such remote proxy repository to function?

**Comment by brunodoamaral** [comment]
I’m now sure, but I think that the solution is to buy an Enterprise Hub license: https://huggingface.co/enterprise

**Comment by julien-c** [comment]
Yes 👍 . And note that we have several JFrog + HF customers so I think the integration works well nowadays

**Comment by tpanza** [comment]
hi @Wauplin , @julien-c ,

Can anyone please clarify the scheduled retirement of `HF_ENDPOINT `? How is integration with JFrog Artifactory supposed to work, without that env var? I noticed that it was dropped from the docs, starting around version 0.19.

I am in a similar position as @Adamwgoh and @h-filzer . In many corporate environments, we must proxy all of our requests to public repos and registries via our Artifactory server. This is for a number of reasons, including internal computing security policies and bandwidth minimization.

(This is analogous to setting `PIP_INDEX_URL` env var, or the equivalent in pip.conf, so that we point to an Artifactory proxy to the PyPI, instead of hitting it directly.)

**Comment by Wauplin** [comment]
Hi @tpanza there is no timeline for this actually. Rest assured we won't remove anything until we have a proper alternative to propose :)
